lib/msal-react/src/useMsalAuthentication.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from "react";
+import { PopupRequest, RedirectRequest, SsoSilentRequest, InteractionType } from "@azure/msal-browser";
+import { AccountIdentifiers, useIsAuthenticated } from "./useIsAuthenticated";
+import { useMsal } from "./MsalProvider";
+import { IMsalContext } from "./MsalContext";
+
+type MsalAuthenticationResult = {
+    msal: IMsalContext;
+};
+
+export function useMsalAuthentication(
+    interactionType: InteractionType, 
+    authenticationRequest?: PopupRequest|RedirectRequest|SsoSilentRequest, 
```

Comment:
Are both `interactionType` and `authenticationRequest` necessary? Can we not infer one from the other?

Reply:
Since types are stripped at runtime and these also share a lot of properties we cannot infer the `interactionType` from just the type of the `authenticationRequest`